### PR TITLE
Add troubleshooting docs

### PR DESCRIPTION
**Pull request by kinow**
To document our findings from today's troubleshooting section in Element with @swzCuroverse. Also added some commands and information that I have used before and may be useful to others :+1: 

Feel free to update this PR as needed to edit the text or add more info.

-Bruno

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -185,6 +190,31 @@ If you are having trouble with links to sections or code blocks, it might
...
+and [`~/.cache/salad`](https://github.com/common-workflow-language/user_guide/issues/268#issuecomment-1276068865).
+You may want to start afresh with a new Python virtual environment,
+installing the dependencies from scratch.
+
+If running `make html` or `make watch` appears to run fine but [gets stuck
+](https://github.com/common-workflow-language/user_guide/issues/268) at some
+part of the process (`10%`, `33%`, `57%`, etc.) for longer than 2-5 minutes
+you may have to try the following:
+
+1. Check if you have the latest versions installed in your Python virtual environment.
```
Comment (mr-c) [inline CONTRIBUTING.md:209]: Latest versions of what?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -185,6 +190,31 @@ If you are having trouble with links to sections or code blocks, it might
...
+2. Use `top` or `htop` to find out if there is any `cwltool` command running. Then
+   try running that command individually to see what is the error message.
+3. Upgrade your version of `docker` if you have [an error message](
+   https://github.com/docker/compose/issues/8121#issuecomment-806055733) similar to
+   `docker: Failed to create the container ID file: open /tmp/....cid: no such file or directory.`.
+4. Try using `udocker` instead of `docker`, by using `pip` to install it, and setting
+   `CWLTOOL_OPTIONS` to `--user-space-docker-cmd=udocker`.
```
Comment (mr-c) [inline CONTRIBUTING.md:216]:
```suggestion
   `CWLTOOL_OPTIONS` to `--udocker`.
```

**Review by mr-c — CHANGES_REQUESTED** [review]
Thanks for consolidating these!

File: CONTRIBUTING.md (hunk 1 above)
Comment (kinow) [reply CONTRIBUTING.md:208]: Ops. Dependencies. Let me update after adding verbosity to make. Thanks!

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -125,6 +125,10 @@ source venv/bin/activate
 # Open <http://localhost:8000/> in your browser
 ```
 
+You can use `udocker` or any other Docker compatible engine to build the
+user guide, by setting the following environment variable:
+`CWLTOOL_OPTIONS=--udocker`.
```
Comment (mr-c) [inline CONTRIBUTING.md:130]:
```suggestion
`CWLTOOL_OPTIONS=--udocker`, or `CWLTOOL_OPTIONS=--podman` for Podman, or `CWLTOOL_OPTIONS=--singularity` for Singularity/Apptainer.
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -185,6 +190,31 @@ If you are having trouble with links to sections or code blocks, it might
 be due to duplicated sections, or to spaces or other characters. To
 preview the generated links, use the `myst-anchors` tool.
 
+### Troubleshooting
+
+General advice is to use the `sphinx-build` command directly instead of `make`:
```
[the remark this answers is not in the document]
Comment (kinow) [reply CONTRIBUTING.md]: Done!

**Comment by kinow** [comment]
Rebased. Thanks for the review @mr-c!

**Pull request merged**